src/xdocs/config_filters.xml

```diff
@@ -999,6 +999,221 @@ public class InputTest {
+            </tr>
+            <tr>
+                <td>files</td>
+                <td>a Regular Expression matched against the file name associated
+                    with an audit event</td>
+                <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                <td><code>null</code></td>
+                <td>8.16</td>
+            </tr>
+            <tr>
+                <td>checks</td>
+                <td>a Regular Expression matched against the name of the check
+                    associated with an audit event.</td>
+                <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                <td><code>null</code></td>
+                <td>8.16</td>
+            </tr>
+            <tr>
+                <td>message</td>
+                <td>a Regular Expression matched against the message of the check associated
+                    with an audit event.</td>
+                <td><a href="property_types.html#regexp">Regular Expression</a></td>
+                <td><code>null</code></td>
+                <td>8.16</td>
+            </tr>
+            <tr>
+                <td>id</td>
+                <td>a string matched against the ID of the check associated with an audit
+                    event.</td>
+                <td><a href="property_types.html#string">String</a></td>
+                <td><code>null</code></td>
+                <td>8.16</td>
+            </tr>
+            <tr>
+                <td>query</td>
+                <td>a string xpath query.</td>
+                <td><a href="property_types.html#string">String</a></td>
+                <td><code>null</code></td>
+                <td>8.16</td>
+            </tr>
+        </table>
+      </subsection>
+      <subsection name="Examples" id="SuppressionXpathSingleFilter_Examples">
+        <p>
+          To configure to reject CyclomaticComplexity errors
+          for all methods with name sayHelloWorld
+          inside FileOne and FileTwo files:
+        </p>
+        <pre>
+&lt;module name="SuppressionXpathSingleFilter"&gt;
+  &lt;property name="files" value="FileOne.java"/&gt;
+  &lt;property name="checks" value="CyclomaticComplexity"/&gt;
+  &lt;property name="query" value="//METHOD_DEF[@text='sayHelloWorld']"/&gt;
+&lt;/module&gt;
+&lt;module name="SuppressionXpathSingleFilter"&gt;
+  &lt;property name="files" value="FileTwo.java"/&gt;
+  &lt;property name="checks" value="CyclomaticComplexity"/&gt;
+  &lt;property name="query" value="//METHOD_DEF[@text='sayHelloWorld']"/&gt;
+&lt;/module&gt;
+        </pre>
+        <p>
+          To suppress checks for all package definitions:
+        </p>
+        <pre>
+&lt;module name="SuppressionXpathSingleFilter&gt;
+  &lt;property name="checks" value=".*"/&gt;
+  &lt;property name="query" query="/PACKAGE_DEF"/&gt;
+&lt;/module&gt;
+        </pre>
+        <p>
+          Suppress checks for parent element of the first variable definition:
```

Comment:
please explain what for in real life it might be required ?
Please give code example for each config block, and show by comment when violation placed without suppression. 

if no reason please substitude it example of suppression by non-query. Your filter is named Xpath, but IT IS NOT LIMITED for Xpath only, it can do filtering/suppressions by all other properties. It is Filter of general purpose + xtpath.

Reply:
> please explain what for in real life it might be required ?

Which part should this explanation be in?  In the description? Is "suppresses audit events for
Checks violations in the specified file, class, checks, message, module id, and xpath"  not the description in "real life"? I cannot understand well, sorry for that.

> if no reason please substitude it example of suppression by non-query.

I'm not sure what the "substitude" means here? Does it simply means that I should "add" an example for suppression with no query?

Thx for your review. I would open up another issue after @rnveach reviews.

Reply:
> if no reason please substitude it example of suppression by non-query. 

done.
> Please give code example for each config block, and show by comment when violation placed without suppression.

Still working on it. I would open up another clean pull request for this.

Reply:
Do not waste time, keep updating code